src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -602,6 +602,23 @@ def trusted_launch_warning_log(namespace, generation_version, features):
             logger.warning(UPGRADE_SECURITY_HINT)
 
 
+def validate_vm_disk_trusted_launch(namespace, disk_security_profile):
+    if disk_security_profile is None:
+        return
+
+    from ._constants import UPGRADE_SECURITY_HINT
+    security_type = disk_security_profile.security_type if hasattr(disk_security_profile, 'security_type') else None
+    if security_type == 'TrustedLaunch':
+        if namespace.enable_secure_boot is None:
+            namespace.enable_secure_boot = True
+        if namespace.enable_vtpm is None:
+            namespace.enable_vtpm = True
+        if namespace.security_type is None:
+            namespace.security_type = 'TrustedLaunch'
```

Comment:
It is mentioned in the feature request #23476 Scenario 1:
OS Disk securityProfile has 1-1 mapping with VM, i.e., OS Disk enabled with Trusted Launch **can only** deploy VM with Trusted Launch.

I think namespace.security_type shoule be set to TrustedLaunch directly without if statement. Hi @AjKundnani could you please help confirm this?
```suggestion
        namespace.security_type = 'TrustedLaunch'
```